app/src/main/java/org/simple/clinic/patient/PatientRepository.kt

```diff
@@ -182,39 +182,41 @@ class PatientRepository @Inject constructor(
         .toObservable()
   }
 
-  override fun mergeWithLocalData(payloads: List<PatientPayload>): Completable {
-    return Single.fromCallable {
-      payloads.asSequence()
-          .filter { payload ->
-            database.patientDao().getOne(payload.uuid)?.syncStatus.canBeOverriddenByServerCopy()
-          }
-          .map(::payloadToPatientProfile)
-          .toList()
-    }.flatMapCompletable(::save)
+  override fun mergeWithLocalData(payloads: List<PatientPayload>) {
+    val payloadsToSave = payloads
+        .filter { payload ->
```

Comment:
We should look into using sequences for more efficient memory operations when refactoring this into a unified sync framework.